src/platform_impl/linux/x11/mod.rs

```diff
@@ -436,13 +442,18 @@ impl<T> EventLoopWindowTarget<T> {
 
 impl<T: 'static> EventLoopProxy<T> {
     pub fn send_event(&self, event: T) -> Result<(), EventLoopClosed<T>> {
-        self.user_sender.send(event).map_err(|e| {
-            EventLoopClosed(if let SendError::Disconnected(x) = e {
-                x
-            } else {
-                unreachable!()
-            })
-        })
+        match self
+            .user_sender
+            .send(event)
+            .map_err(|e| EventLoopClosed(e.0))
+        {
+            Ok(_) => {
+                // Can't send the event as an error, so panic
+                self.waker.wake().unwrap();
+                Ok(())
+            }
+            Err(e) => Err(EventLoopClosed(e.0)),
+        }
```

Comment:
I think you can use `Result::and_then` here instead? Or maybe that requires [RFC 2229](https://github.com/rust-lang/rfcs/blob/master/text/2229-capture-disjoint-fields.md)?

Reply:
I tried to follow your advice and use `Result::and_then` but clippy yelled at me and said I should use `Result::map` instead. I don't think that RFC 2229 applies here.